packages/component-library/src/components/form/mt-number-field/mt-number-field.vue

```diff
@@ -252,6 +269,38 @@ export default defineComponent({
       },
       immediate: true,
     },
+
+    min: {
+      handler(newMin: number | null) {
+        if (this.currentValue === null || this.currentValue === undefined) {
+          return;
+        }
+
+        const currentNumericValue = Number(this.currentValue);
+        if (newMin !== null && !Number.isNaN(currentNumericValue) && currentNumericValue < newMin) {
+          this.computeValue(newMin.toString());
+          this.$emit("update:modelValue", this.currentValue);
+        }
```

Comment:
I'm not a fan of implementing the bounds check again, why not:
```suggestion
        if (Number.isNaN(Number(this.currentValue))) {
          this.computeValue(currentValue.toString());
          this.$emit("update:modelValue", this.currentValue);
        }
```